lib/src/chatbox.dart

```diff
@@ -296,17 +296,23 @@ class ChatBoxState extends State<ChatBox> {
       },
       shouldOverrideUrlLoading: (InAppWebViewController controller,
           NavigationAction navigationAction) async {
-        if (navigationAction.navigationType == NavigationType.LINK_ACTIVATED) {
-          if (widget.onUrlNavigation != null) {
-            // The onUrlNavigation function has been defined, so let's see if we should open the browser or not
-            if (widget.onUrlNavigation!(UrlNavigationRequest(
-                    navigationAction.request.url!.rawValue)) ==
-                UrlNavigationAction.deny) {
-              return NavigationActionPolicy.CANCEL;
-            }
+        if (Platform.isAndroid ||
+            (navigationAction.navigationType ==
+                NavigationType.LINK_ACTIVATED)) {
+          // NavigationType is only present in iOS devices (Also MacOS but our SDK doesn't support it.)
+
+          final webUri = navigationAction.request.url ?? WebUri("about:blank");
+
+          // If onUrlNavigation is null we default to allowing the navigation request.
+          final urlNavigationAction = widget.onUrlNavigation
+                  ?.call(UrlNavigationRequest(webUri.rawValue)) ??
+              UrlNavigationAction.allow;
+
+          if (urlNavigationAction == UrlNavigationAction.deny) {
+            return NavigationActionPolicy.CANCEL;
           }
 
-          if (await launchUrl(navigationAction.request.url!)) {
+          if (await launchUrl(webUri, mode: LaunchMode.externalApplication)) {
```

Comment:
Good catch.
I don't know why `LaunchMode.externalApplication` isn't the default. I mean, url_launcher is meant to launch a browser, why do other options even exist? WTF ?!

Reply:
The default for iOS and Android, launch web URLs inside the webview and then treats all other urls as opening in exeternal applications.

Reply:
I agree that it makes more sense if the default were to open in an external application.